### ISSUE TITLE
Something went wrong – export takes long time and eventually fails

### ISSUE THREAD

**Issue by iby** [Gifox/Tracker#159]
**🙏 We're aware that this is happening and this is the top priority issue for us. We want this fixed as much as everyone else and kindly ask to not be an 🍑 with "+1" and "me too" here. If you want to help, please take a moment and provide the details that can help with nailing this down. All info below.**

**TO FIX THIS:** Restart the app, open editor (rollover the recording and press second button), press `Done` to re-export. Also make sure you use the latest version and consider helping us with details! 👇

Unfortunately we don't know what's causing this and we can't reproduce the issue locally. Some users indicated that this happens when Mac runs on low battery. Some said that this happens when Mac awakes from sleep. There are reports that this happens after connecting an external monitor.

> 💡 The overall suspicion is that this has something to do with graphic processor being reconfigured and Gifox loosing touch with it. This goes in line with all three theories.

We very much need additional input with any hints and observations, especially if they confirm low battery / sleep / external monitor theories above:

> ⁉️ We need to figure out the exact steps to reproduce the issue to be able to understand out what is causing it.

1. Immediately after the export fails use [sysdiagnose](http://hints.macworld.com/article.php?story=2014032113470243) to create a detailed system report:
   a. Briefly press the following keys simultaneously to trigger a sysdiagnose from the Finder:   <kbd>Command</kbd> + <kbd>Option</kbd> + <kbd>Shift</kbd> + <kbd>Control</kbd> + <kbd>Period (.)</kbd>. You should see the screen flashing briefly, which indicates the sysdiagnose has started gathering data.
   b. The sysdiagnose process can take 10 minutes to complete. Once finished, the folder `/private/var/tmp` should appear automatically in the Finder and the sysdiagnose file there will look similar to `sysdiagnose_2020.08.13_08-37-52+….tar.gz`.
   c. The generated file can be 300-400 MB in size and can contain some private details, so the best thing to do is to use a service like [WeTransfer](https://wetransfer.com) or anything else of your choice to send a link to team@gifox.io.
2. Send us the diagnostics data (library window > bottom-right menu button).
3. When the export fails don't quit the app and try:
   a. Open any earlier successful recording in the editor. Do you see the image or does it come out black? Can you play it?
   b. Press `Done` to re-export it, does the export succeed?
   c. Record a short video with the standard screen capture and import it into Gifox. Does the conversion succeed?
4. If you use external monitor: disconnect it and relaunch the app. Make a quick recording, reconnect it and make another recording. Disconnect it and make another recording. Did anything go wrong?
5. If you have any additional details or comments, please let us know!

✌️

**Update Jul 19:** This was significantly improved in version 2.2.1 but not fixed. Unfortunately, this won't be fixed in the upcoming 2.2.2 release either – the update is focused on better diagnostics for this issue and will deliver a few other improvements. Nevertheless, this is the top-priority task.

**Update Aug 13:** We've narrowed down the problem to a macOS system component and speaking with Apple technical support to find a solution. Unfortunately, there are no immediate workarounds and we need more detailed info for this to move forward. I've updated the steps above and would appreciate if you can provide the **sysdiagnose** data from #&#x2060;1.

**Update Aug 24:** Please try the latest 2.2.3 release! 🤞

**Comment by wtfluckey** [comment]
🙋‍♀️  This started happening for me today. The app was working a few days ago with no issue but started acting up today. My battery is full, the laptop did wake from sleep this morning and I saw the issue after I tried to use Gifox for the first time today, but it had been awake for hours. The first capture that failed was a rather large and long capture, I thought maybe that was the issue so I tried with shorter and smaller ones, but still no luck. 

Thanks for your help with this! If there's any other info I can provide, let me know. I'm happy to help!

Here are the logs and settings files:
[App Settings.txt](https://github.com/Gifox/Tracker/files/4927227/App.Settings.txt)
[App Logs.txt](https://github.com/Gifox/Tracker/files/4927226/App.Logs.txt)

![Screen Shot 2020-07-15 at 1 29 40 PM](https://user-images.githubusercontent.com/2893463/87581863-3d5fa700-c69f-11ea-9ec5-1150d3ebdae0.png)

**Comment by ianbytchek** [comment]
I added 2 (import and edit) checks to the original post. When you run into this issue again, can you test it please and share the results? 👍

**Comment by ekilah** [comment]
I've been having this issue consistently since my upgrade to Gifox 2

[App Logs.txt](https://github.com/Gifox/Tracker/files/4969739/App.Logs.txt)
[App Settings.txt](https://github.com/Gifox/Tracker/files/4969740/App.Settings.txt)

I actually can't figure out how to "ignore the issue and open the editor" right after the issue happens, since if I click the "Ignore" button, the gif's row in the popup goes away. If I quit and restart Gifox, though, I am able to click the editor button, and it does appear to show the contents of the gif in the editor view just fine.

![image](https://user-images.githubusercontent.com/3213339/88352250-02e4b280-cd0e-11ea-881c-dd860f49cbef.png)

In fact, if I click "done" in the editor window, the failed gif seems to re-save itself and that does succeed, so I guess that's a workaround we can use for now while this is figured out.

And I tried the regular Mac screen recording, and dragged it onto the Gifox toolbar, and that seemed to convert just fine.

**Comment by ekilah** [comment]
Update: My computer is plugged in and at full battery most of the time, so that isn't part of it.

I put the computer to sleep and woke it back up, that didn't seem to cause issues.


However, as soon as I plugged in my external monitor (4K over Thunderbolt 3), the issue re-presented itself. Recordings on either my laptop display or my external display failed as soon as I plugged it in, when it was working seconds before I plugged it in.


And this time I didn't restart Gifox before trying the mp4-drag-to-convert, and that also failed. So, sorry for the report above.  Unlike the other rows, though, which still disappear when I click "Ignore" on them, this dragged-video conversion failure stays around when I click ignore on it. When I try to look at it in finder, it points me to a 0KB file it created, which is obviously not playable. Opening the editor works, and I can see the video, but clicking "Done" retries the conversion, which fails again.

**Comment by juniorclearmetal** [comment]
Same issue here.

> If you experience this issue, please send us your diagnostics data (library window > bottom-right menu button).

[App Logs.txt](https://github.com/Gifox/Tracker/files/4969909/App.Logs.txt)
[App Settings.txt](https://github.com/Gifox/Tracker/files/4969910/App.Settings.txt)

> Please check the following and let know what happens:
> a. Ignore the error and open the editor. Do you see the image or does it come out black?

Sorry, not sure what editor this refers to

> b. Record a short video with the standard screen capture and import it into Gifox. Does it also fail to convert?

Yes

> If you have any additional details or comments, please let us know!

Quitting and restarting often helps. Once I get one with "Something went wrong" even a 1 second tiny capture fails with the same.

**Comment by ianbytchek** [comment]
Everyone, thank you for the input! Great point about the `Ignore` button behavior. It comes from pre-2 versions where the best thing was to remove the recording since there was no re-export. We'll get this fixed in the upcoming release.

@ekilah Wow, that's interesting! Especially where you say that export fails but you can see the image in the editor. Export and playback (image in the editor) use the same tools to create the graphics. If one works and one doesn't it's a pretty big deal. Can you check this one more time with updated step 3 and then step 2 instructions at the top?

@juniorclearmetal I updated the top paragraph in the original post with better instructions. Yes, unfortunately this has nothing to do with the size or duration. Once it fails it stays this way. Do you use an external monitor by any chance?

P.S. The original post is updated with new details and checks.

**Comment by juniorclearmetal** [comment]
@ianbytchek thanks, I'll pay attention to the new details and try to provide more info. I can say this much now:

- I use an external monitor. Next time this happens I'll try the unplug, etc. and let you know
- my Mac is always plugged into power, so low power not relevant
- I don't _think_ it's related to sleep at all but I'll watch for that

On the positive side this led me to find out Gifox has an editor. :-/

**Comment by ranihorev** [comment]
Gifox 2 got to the point where it's almost unusable. Every other save fails. I restart the app, recapture and it works for a couple of times, and then the same issue returns
![image](https://user-images.githubusercontent.com/17147717/88573433-94e20900-cff5-11ea-9cc0-d93853aaa7b6.png)

**Comment by ianbytchek** [comment]
Apologies for the troubles @ranihorev. Unfortunately this isn't an easy fix and happens somewhere in closed Apple's framework code. The fact that the issue isn't easily reproducible also doesn't help and we're still gathering intel on it. I know this is the pain in the ass, but it would be great if you can provide any actual details that might help. All the details are kept up to date at the [top in the issue](https://github.com/Gifox/Tracker/issues/159#issue-642137883). It sounds like it doesn't take long before it returns after you relaunch the app, perhaps it has something to do with the second monitor?

**Comment by ranihorev** [comment]
@ianbytchek can I revert back to Gifox 1? Do you support OSX 10.15?

Yes, I have two monitors, and I also connect and disconnect them quite often. They also have a different `devicePixelRatio` if it matters

**Comment by ianbytchek** [comment]
The fresh build v2.2.2 was published yesterday (Mac App Store version is still in review, will arrive in a day or two). Besides other improvements it attempts to solve this issue and gathers additional details if it occurs. You probably won't see any problems for a few days (as before) but this starts happening (very likely) please be sure to send your diagnostics data so we evaluate everything one more time with new data.

@ranihorev It has [some problems (and workarounds)](https://github.com/Gifox/Tracker/issues/115) on Catalina, a few people still use it, but it's far from fun. We had plans to patch-update the original app to address Catalina issues and include other fixes from version 2, but had a few setbacks, including this issue. Our primary focus remains version 2 improvements and ongoing development, so, there's not ETA and officially version 1 won't be maintained in a longer term.

**Comment by iliya-yordanov** [comment]
On macOS Big Sur beta 4, I get this error all the time :(

**Comment by ianbytchek** [comment]
@iliya-yordanov please send the diagnostics data. If you can do any tests from above, would be super helpful!

**Comment by ekilah** [comment]
Occurred again for me on 2.2.2

**If you experience this issue, please send us your diagnostics data (library window > bottom-right menu button).**
[App Logs.txt](https://github.com/Gifox/Tracker/files/5052464/App.Logs.txt)
[App Settings.txt](https://github.com/Gifox/Tracker/files/5052465/App.Settings.txt)





**When the export fails don't quit the app and try:
a. Open any earlier successful recording in the editor. Do you see the image or does it come out black? Can you play it?**

No. They all look like this. The timeline on some seems to render a few frames, but regardless they won't play.
![image](https://user-images.githubusercontent.com/3213339/89814812-954fc900-daf8-11ea-9764-c258caf4bdf9.png)


**b. Press Done to re-export it, does the export succeed?**
To be clear, `it` here is a previously-working export, as in `(a)` above.

No. At first my previously-working export hung like this after clicking `Done`:
![image](https://user-images.githubusercontent.com/3213339/89815026-f2e41580-daf8-11ea-91fb-3c2d62309330.png)

After 20 seconds or so, it eventually changed to the failure state, just like the new recordings do:
![image](https://user-images.githubusercontent.com/3213339/89815151-232bb400-daf9-11ea-812b-284f74001d81.png)

**c. Record a short video with the standard screen capture and import it into Gifox. Does the conversion succeed?**

The video worked (cmd+shift+6 to choose the Mac native video recording since I use 5 for Gifox) but dragging it onto the Gifox menu bar and dropping it in the conversion box failed. The progress bar did move quickly at first but failed at about the 70% mark.


**If you use external monitor: disconnect it and relaunch the app. Make a quick recording, reconnect it and make another recording. Disconnect it and make another recording. Did anything go wrong?**

Nothing went wrong. I did several sleeps and disconnects to try and reproduce it, but it just won't reproduce for me.

Experiencing issue
Disconnect monitor
Relaunch Gifox
Record ✅ 
Reconnect monitor
Record ✅ 
Disconnect monitor
Record ✅ 
Connect monitor
Sleep, wait 20 seconds
Wake
Record ✅ 
Disconnect monitor
Record ✅ 
Sleep, wait 20 seconds
Wake
Record ✅ 
Connect monitor
Record ✅ 

None of these steps ran into issues. And yet, it happens for me all the time, somehow. Maybe it has more to do with how long the app is running? I basically never shut my laptop down, like once a month. Usually it just sleeps. Or, maybe I didn't sleep long enough (the computer didn't lock, it just went to sleep and woke up immediately when I pressed a key, no login screen required, which usually happens after 5min of sleep w/ my settings)

**If you have any additional details or comments, please let us know!**

Battery: 63%, charging (don't think this is relevant, happens all the time at 100% charging)
External monitor: 4K over Thunderbolt 3 and via a dock, recently plugged in
Sleep: recently woken from sleep

**Comment by ianbytchek** [comment]
@ekilah can't thank you enough, this is brilliant!!! Unfortunately, this 💯 confirms one of the theories: [AVVideoPlayer](https://developer.apple.com/documentation/avfoundation/avaudioplayer) component that Gifox heavily relies on is failing. I'll have to take this to Apple engineering support to try to understand why this can be happening and if there's a quick fix. In the meantime, we'd also need to look for alternative solutions. We had an idea floating around but it was a little pointless implementing it without understanding the exact source of the problem. Especially considering it's a major rewrite and deviation from how our video processing works. At least now it makes sense to open this can of worms. Again, thanks for the detailed research – you're the best!

**Comment by ekilah** [comment]
Glad to help. Sorry it took a while to try again. Let me know if more testing becomes useful, I'll do what I can. Love the tool, so I want it to work nicely again :)

**Comment by drulang** [comment]
![image](https://user-images.githubusercontent.com/2532229/89949526-06c17180-dbf6-11ea-911b-fbdf50a8b8b8.png)

**Comment by ianbytchek** [comment]
Hello everyone! Again, apologies for the lack of visible progress. We narrowed down the problem to a macOS system component and speaking with Apple engineering support to find a solution. Unfortunately, there are no immediate workarounds and we need more detailed info for this to move forward. I've updated the steps in the original post and would appreciate if you can provide the sysdiagnose data from #⁠1. Immediately after the export fails use [sysdiagnose](http://hints.macworld.com/article.php?story=2014032113470243) to create a detailed system report:

1. Briefly press the following keys simultaneously to trigger a sysdiagnose from the Finder:   <kbd>Command</kbd> + <kbd>Option</kbd> + <kbd>Shift</kbd> + <kbd>Control</kbd> + <kbd>Period (.)</kbd>. You should see the screen flashing briefly, which indicates the sysdiagnose has started gathering data.
2. The sysdiagnose process can take 10 minutes to complete. Once finished, the folder `/private/var/tmp` should appear automatically in the Finder and the sysdiagnose file there will look similar to `sysdiagnose_2020.08.13_08-37-52+….tar.gz`.
3. The generated file can be 300-400 MB in size and can contain some private details, so the best thing to do is to use a service like [WeTransfer](https://wetransfer.com) or anything else of your choice to send a link to team@gifox.io.

We will be forwarding this to Apple engineering support for further investigation. If you're interested in any technical details please drop us email and I'll be happy to answer any questions.

**Comment by bernardorubin** [comment]
I can't record anything, this is a really big issue for me, please fix this

**Comment by drulang** [comment]
Can we expect a refund if this cannot be fixed in a reasonable amount of time?

**Comment by ianbytchek** [comment]
We have started rewriting the video processing part earlier this week where the export fails. Hopefully there will be a beta ready next week, no solid ETA, but like said, this is the top priority.

This \***most likely**\* will solve the issue with exporting, but there's no technically reasonable way for us to not use the failing [AVPlayer](https://developer.apple.com/documentation/avfoundation/avplayer) component without \***loosing the editor**\* – it is needed for previewing the composition. My biggest concern is that if we don't provide the requested information to Apple this simply won't be solved.

I wouldn't ask if this wasn't critical, please [gather and send sysdiagnose](https://github.com/Gifox/Tracker/issues/159#issuecomment-673368834) report and help me to properly fix this, otherwise this will simply return in a different issue… 🙏

**Comment by adam-zethraeus** [comment]
FYI seeing this on Big Sur 11.0-beta (20A5343j)

**Comment by ianbytchek** [comment]
Hello everyone! 👋 Just published the freshly baked version 2.2.3 that supposed to resolve this issue, Mac App Store and Setapp are coming shortly once reviewed by those teams. 

Gifox no longer uses the failing component for the export and we expect this to be fixed. As mentioned, because the issue is limited to certain devices and none out of 5 that we have access to can reproduce this problem there is a chance that something else might require additional fine-tuning. We've tested the release with 10.12 Sierra – 11.0 Big Sur macOS versions and things look so far so good. Please let me know if something doesn't act normal? Again, you probably won't see any problems until the app runs in the background for a couple of days.

P.S. You are likely to notice sub-optimal performance during editing, this is expected as we didn't have much time to optimize the new build and in the next release or two will try to improve that part as well! ✌️

**Comment by ranihorev** [comment]
Now the app is completely unusable. I can't save anything.
It gets stuck at the beginning of the export...

**Comment by ianbytchek** [comment]
@ranihorev Please provide diagnostics data, you can use in-app contact button to send it securely if don't feel like posting here. Can't help you without it.

**Comment by jackcanta** [comment]
I recently updated as well and it's still broken.  I'll try sending some diagnostics.

**Comment by ackernaut** [comment]
I’m not sure if this is the same issue. I recently installed version 2 and it will export gifs unless I customize the export settings, then the export stops working. If I restart and press done from the timeline editor it still does not export. So for now, I’m rolling with default settings 🏄‍♂️

**Comment by ianbytchek** [comment]
Hey @ackernaut, can you please send over the diagnostics data? You can do that directly from the app using the message button in the bottom right corner. I'll able to tell more once I can see what's going on behind the scenes.

**Comment by ianbytchek** [comment]
**Update:** Good news and bad news. Good: so far there are no signs of the original error. Bad: I had a chance to look at the fresh set of diagnostics data, thanks @ackernaut and @ranihorev! It seems there's now a different issue where export never completes or takes (literally) ages. It's not not fully clear what exactly is happening, once I have more details I'll create a separate issue. Will post updates here as have any new info. If you're experiencing any issues with the new version, please do let us know.

**Comment by ianbytchek** [comment]
Let's check something. Thumb up this comment if you're getting this error while using a Dell monitor. Thumb down if not. ✊

**Comment by ekilah** [comment]
BenQ monitor here. Gifs fail now immediately after restarting the app too on the latest update, which means it's completely unusable now.

**Comment by ianbytchek** [comment]
Okay, so it's not just Dells… 😔 Thanks for the input. For anyone finding the same behaviour please consider going back to 2.2.2 while we sorting this out, it's available [here](https://distribution-absf.s3.eu-central-1.amazonaws.com/gifox/020202.00.dmg).

**Comment by jynnie** [comment]
MSI monitor here. Gif fails immediately on restart of the app; but if I restart and edit the gif, it exports perfectly fine. Version 2.2.3. Gif fails don't happen consistently.

**Comment by ianbytchek** [comment]
**Update:** Posting a quick TL;RD as well as the latest technical discoveries for those who're helping or interested.

**TL;DR:** We've put together the new pre-release which collects extra diagnostics data during recording and export. Now exports that were hung up will fail after a few seconds if they cannot be processed, like they did in 2.2.2. Once it does please use the in-app messaging to email anonymized diagnostics data. This release is not available via regular update, please [download](https://distribution-absf.s3.eu-central-1.amazonaws.com/gifox/020203.01.zip) and install it manually. If you have any questions, please feel free to email me.

After going through more diagnostic data we're back to [square one](https://github.com/Gifox/Tracker/issues/159#issuecomment-671525747) where the problem was narrowed down to system components. In 2.2.3 we moved away from AVPlayer to a more primitive and less error-prone AVAssetReader. Both are system components and come from the same framework. While 2.2.3 fixed the problem for some, it turned out to be useless for others. Both builds are failing to read the recorded/imported video, hence we're getting a failure in 2.2.2 and stuck progress in 2.2.3.

The problem is probably not as much with Dell monitors nor color profiles as suspected earlier, but with the dual-GPU graphic cards and discrete GPU mode with 30-Bit Color support. This is the one common thing that stands out across all available reports. If you haven't sent us a system report but interested in helping out you can check your [System Report](https://support.apple.com/en-gb/HT203001) > Graphics/Displays > PCIe Bus > Framebuffer Depth and see if it says `30-Bit Color (ARGB2101010)` and upvote this with 🎉 emoji to confirm or 😕 emoji to contest, it would look something like this:

![image](https://user-images.githubusercontent.com/1086845/91839224-a2b72980-ec46-11ea-80a3-49bde8f85bf5.png)

We have received a detailed sysdiagnose report and will resume talks with Apple in hopes the AVFoundation team responsible for the aforementioned components can help us out. It's also a little easier to fiddle with AVAssetReader component, so we will continue posting private builds here to try out different configurations.

That's the plan. Again, thanks everyone who's helping with this and feeding us the info! ✌️

**Comment by atymic** [comment]
> MSI monitor here. Gif fails immediately on restart of the app; but if I restart and edit the gif, it exports perfectly fine. Version 2.2.3. Gif fails don't happen consistently.

Same here, asus monitor.

**Comment by amongmany** [comment]
I'm on an apple monitor. This issue is getting so bad that I'm on the brink of looking for a new app for screencaps. I can confirm that the info above `30-Bit Color (ARGB2101010)` applies to me if that helps.

**Comment by mikecules** [comment]
How does one get the version 2.2.3 of this app who have purchased it through the Apple App Store? @ianbytchek

**Comment by ianbytchek** [comment]
Thanks everyone for confirming the [30-Bit Color (ARGB2101010)](https://github.com/Gifox/Tracker/issues/159#issuecomment-684799768) discovery above. The engineering support at Apple confirmed they're looking into our reports and will need sometime to digest everything. In the meantime we continue to tweak things that might be related to the issue and releasing the [2.2.4 public beta](https://gifox.io/prerelease). You can also get it by enabling pre-releases (if you have direct-download version) and checking for updates. If you don't have the direct-purchase license but want to help with testing please email us at team@gifox.io and we'll set you up.

We plan to test out several theories with this and upcoming betas, so we'll be releasing more builds this week and I will publish further updates here. ✌️

**Comment by andym03** [comment]
[App Logs.txt](https://github.com/Gifox/Tracker/files/5309554/App.Logs.txt)
[App Settings.txt](https://github.com/Gifox/Tracker/files/5309555/App.Settings.txt)

Not entirely sure if this is the same issue, but thought i'd throw some logs down to see if it helps. For some reason recording my JIRA bug failed. Macos Catalina, paid Giffox version.

Good luck and thanks guys, love the app otherwise.

**Comment by jkotoff-ctr-asp** [comment]
Nothing special about my setup. Just a macbook and no external monitors.  
Worked when I installed it a few weeks ago, updated this morning and recordings fail.  
[App Settings.txt](https://github.com/Gifox/Tracker/files/5356342/App.Settings.txt)
[App Logs.txt](https://github.com/Gifox/Tracker/files/5356340/App.Logs.txt)

**Comment by ianbytchek** [comment]
I feel like this is never going to end… 😖🔫  Sorry for the f-up everyone, really am! :(

The new build just been published with these new failures addressed. It's also supposed to addresses this very issue, please give it a go and let me know how it looks. And thanks to @jynnie and everyone else who keeps feeding in the data – it's the only thing that makes the real difference and really helping understanding what's going wrong.

**Comment by ianbytchek** [comment]
I just received a few reports that there's still a problem and something is definitely screwed up. It's not very clear what, so I just reverted everything back to 2.2.3, please get it from the [website](https://gifox.io) and I'll try to figure out the cause of this over the weekend. If you can send your diagnostics data (bottom right button) it would be very helpful. Again, apologies for the troubles… 🤦‍♂️

**Comment by angeliquejw** [comment]
I've been using the Setapp version of Gifox and had no issues until today (and downgrading via the web version worked, but now I don't have the benefits of the paid version any more).

I encounter the error when attempting to record a Firefox browser window. I'm using an external monitor, but was recording on my MBP's display (not the monitor). 

[App Logs.txt](https://github.com/Gifox/Tracker/files/5359878/App.Logs.txt)
[App Settings.txt](https://github.com/Gifox/Tracker/files/5359879/App.Settings.txt)

🍀 Good luck with debugging.

**Comment by JanC** [comment]
I'm a Setapp user and it started happening on both of my macs as well. Is there anything I can do to help debugging besides attaching the logs?

> When the export fails don't quit the app and try:> 
> a. Open any earlier successful recording in the editor. Do you see the image or does it come out black? Can you play it?

Yes an earlier successful recording can be opened in the editor and it plays.

> b. Press Done to re-export it, does the export succeed?

No. This also fails in the same way a new recording does.

> c. Record a short video with the standard screen capture and import it into Gifox. Does the conversion succeed?

This also fails.


> If you use external monitor: disconnect it and relaunch the app. Make a quick recording, reconnect it and make another recording. Disconnect it and make another recording. Did anything go wrong?

No, same behaviour in all steps.


Good luck 🤞

[App Logs.txt](https://github.com/Gifox/Tracker/files/5363515/App.Logs.txt)
[App Settings.txt](https://github.com/Gifox/Tracker/files/5363516/App.Settings.txt)

**Comment by ianbytchek** [comment]
We've published the fresh Setapp release (@angeliquejw, @JanC), it's currently in review and should be available in an hour or two.

The direct-download version was also just released, please check for updates and let me know if something ain't back to normal.

Thank you for quick feedback everyone! 👍

**Comment by JanC** [comment]
I can confirm that the fix works (using `2.2.4 4` from Setapp) 
thanks 👍

Just out of curiosity .. what was the issue?

**Comment by ianbytchek** [comment]
@JanC thanks for verifying! It was a double regression in a large update addressed improve and stabilize export handling to resolve #159. 

First, we've missed verification of an intermediary GIF palette folder during export. unit tests handle that separately and integration tests passed because it was already there from 2.2.3 build. The second issue turned out to be with handling timestamps between different frameworks. It was never the issue as it was handled out of the box in earlier approach and was tricky to catch because it required a fairly long or high-fps export – none of which is used in unit tests for time and performance reasons.

But most importantly, the issue was with breaking the golden rule of never doing Friday eve releases… 🤦‍♂️

**Comment by ianbytchek** [comment]
Ooops, I thought I was commenting on #175 and closed the wrong issue – the above comment applies there. But like said, the 2.2.4 release was meant to resolve this issue too.

The issue that was causing the exports to fail remains on the Apple side and unknown. From @jynnie feedback and some other logs I was able to pin point the intermediary file that was failing to process and restructured the part responsible for encoding to work without it. The [closest details](https://stackoverflow.com/q/33118538/458356) that I could find refer to a similar problem with some of the recorded files failing to read. There's an available sub-optimal solution for that, however, that doesn't solve the issue of failing playback inside the editor that was confirmed by @ekilah in the [feedback](https://github.com/Gifox/Tracker/issues/159#issuecomment-671513648) above.

So, I will reopen this issue for the time being and would be grateful if you can continue watching this and providing feedback. We have added more logging and tracing around media failures, so once you hit any recording or export issues please do send your logs – they will to contain exact error codes and we can figure out how to work around the new (old) issues.

**Comment by ekilah** [comment]
The progress bar now gets to about 2/3 full before failing, but it still fails consistently on 2.2.4 for me today:

[App Logs.txt](https://github.com/Gifox/Tracker/files/5375006/App.Logs.txt)
[App Settings.txt](https://github.com/Gifox/Tracker/files/5375009/App.Settings.txt)


Quitting and restarting, opening the editor shows the recording just fine, but clicking "Done" there does not get the export to work like it used to after an app restart, so this is less usable for me now than the 2.2.2 revert has been where I can at least do that.

**Comment by ianbytchek** [comment]
Thanks for the feedback @ekilah! It looks like you could have been using 2.2.4.03 and not the most recent 2.2.4.04 build. The latest error messages in logs is exactly what 2.2.4.04 addresses, I mentioned that in the [comment](https://github.com/Gifox/Tracker/issues/159#issuecomment-707207993) above. Can you try getting the latest build directly from [here](http://distribution-absf.s3.eu-central-1.amazonaws.com/gifox/020204.04.dmg) (no need to remove the previous version, you can extract it into downloads and try it there).

However, in theory that error is still possible due to GIF limitations and would be caused by a very high frame rate nearing to 100, which you would get when setting export frame rate to `50` and speed multiplier to `2`. The immediate fix would be to reduce export frame rate or export speed. That doesn't appear to be the case with your configuration, hence I'm assuming you could have been using the older build.

I will look into how we can work around this limitation further. If you have a chance, please let me know whether this happens with the usual values when `export frame rate` × `speed multiplier` < `50` in the latest build, in that case the problem might be with the raw file itself that is being converted, which would require a different workaround. Again, thank you for the continued feedback! 🙏

**Comment by ekilah** [comment]
I am all but certain I already had `2.2.4.04` yesterday, but I tried again just to make sure with your link, same error (you should consider adding that last build number `.04` in the diagnostic data logs?)

![image](https://user-images.githubusercontent.com/3213339/96026028-d58a3b00-0e0a-11eb-90db-262a528d3c83.png)

![image](https://user-images.githubusercontent.com/3213339/96026066-e175fd00-0e0a-11eb-96f0-552e4ed1e853.png)

[App Logs.txt](https://github.com/Gifox/Tracker/files/5379972/App.Logs.txt)
[App Settings.txt](https://github.com/Gifox/Tracker/files/5379975/App.Settings.txt)

As you said, my settings don't seem to indicate an issue like you described:
![image](https://user-images.githubusercontent.com/3213339/96025965-be4b4d80-0e0a-11eb-990f-3c0f7086f637.png)

**Comment by tomsoderlund** [comment]
Just bought the app, have the same problem too. v2.2.4 on MacBook Air 2019, macos 10.15.7.

**Comment by jynnie** [comment]
Hey! Recently updated to v2.2.4 and getting a similar but slightly different issue occasionally. Before the new update, GIFs would get corrupted but if I restarted the app and edited and re-exported, could recover. Now I'm getting the occasional issue of no error and the preview image shows up; but the outputted GIF is just black. In edit, it also shows only black frames.

![image](https://user-images.githubusercontent.com/7284723/96929495-35768680-1488-11eb-864b-5ba4c5891d59.png)

![image](https://user-images.githubusercontent.com/7284723/96929584-550daf00-1488-11eb-9457-c5859d9c570d.png)

But still, if I restart gifox, open edit, and re-export -- the GIF is fixed.

**Comment by ianbytchek** [comment]
Hello everyone. Overall it seems that there's a lot fewer of these requests, I feel we're slowly getting there.

We have the new 2.2.5 Beta ready. I meant to get back to all of you earlier but a few other improvements took a bit longer than planned. The issue with the failing export as reported by @ekilah must be addressed, but there's a chance that there's something else causing it. In any case, now there's even more detailed logging in place, so if something goes wrong we should easily trace it down once you send the logs.

@ekilah Thank you for the detailed input. Yes, missing build number was the long-timer that was lost along the way – fixed now.

@tomsoderlund Sorry for the lagged reply. Like said above, this should be fixed in Beta, please give it a go and send logs if something ain't working. If the issue remains, please revert to [2.2.2](https://distribution-absf.s3.eu-central-1.amazonaws.com/gifox/020202.00.dmg) until the next build. If you're on App Store version please [email](mailto:team@gifox.io) us and we'll set you up with the regular license.

@jynnie Thank you for verifying and the details. Yes, unfortunately that proves the original problem is still there. The thumbnail image gets generated at the start, so it isn't really relevant. But the fact that the output, the preview and the timeline are all rendered blank with no failures indicates that something is wrong with the system framework. When this happens next time, could you check if the same happens to old recordings, i.e., ones the app was launched/relaunched.

Thank you everyone and I look forward to further updates on this! 🙏

**Comment by ekilah** [comment]
Do you have a new direct download link for us to try?

**Comment by ianbytchek** [comment]
Ah sorry, yes, it would replace your current build via a regular update. Yes, it's available [here](https://distribution-absf.s3.eu-central-1.amazonaws.com/gifox/020205.00-beta.dmg).

**Comment by jynnie** [comment]
> When this happens next time, could you check if the same happens to old recordings, i.e., ones the app was launched/relaunched.

Yes, will do! Thanks for all the work and quick responses

**Comment by jynnie** [comment]
> > When this happens next time, could you check if the same happens to old recordings, i.e., ones the app was launched/relaunched.

Hey, can confirm previous recordings that succeeded also show up all black.

**Comment by wmhilton** [comment]
I'm happy to report it has been working for me lately now!

**Comment by agileurbanite** [comment]
I'm on 2.2.5 and experiencing this as well.  This is a workaround but looking forward to a fix

**Comment by wtfluckey** [comment]
Just ran into the black frame issue for the first time. Downloaded the beta and it's working again!

**Comment by oleg-slapdash** [comment]
I am 2.2.5 and still experience this. My gifs just end up black without any errors. This is very disappointing.

**Comment by mtrofimiuk** [comment]
2.2.5 on Big Sur, no issues so far.

**Comment by ivanpk** [comment]
2.2.5 on OS X 10.15.6, I do have this issue.

**Comment by b6pzeusbc54tvhw5jgpyw8pwz2x6gs** [comment]
2.2.5 on OS X 10.15.6, I do have this issue.

**Comment by austinbiggs** [comment]
2.2.5 on 10.15.7 (Catalina), getting all black gifs. $15 for a piece of software that can't perform it's sole purpose... 🤨

**Comment by daanishnasir** [comment]
happening to me on Big Sur. Everything is black.

**Comment by andym03** [comment]
2.3.0-beta.0, macOS Catalinam, still having this issue :( It seems to be somewhat reliably fixed by restarting giffox, which is better than it used to be.

**Comment by iby** [comment]
Hello everyone! This issue is so old and big – it smells of desperation… 😖

We had another big rework of how things are captured and processed, which monumented in yesterday's 2.4.2 release. Could you give it a go and let me know if anything similar is still going on?

**Comment by iby** [comment]
Hello friends! I didn't see any relevant reports in a long while, I believe this was eventually fixed somewhere deep down in macOS, so I'm going to close this down. If anyone runs into this, please leave a comment! 🙏

**Comment by ekilah** [comment]
👍 I haven't seen it in a long time.

**Comment by pambram** [comment]
I still get black screen recordings to this day. 2.6.1 on 13.4.1. 

If it's any consolation, I found that it happens less often if I select exactly these options
<img width="580" alt="image" src="https://github.com/Gifox/Tracker/assets/479048/a2659dd0-ac5c-4d8b-8837-308cf2742288">

Any other combination does not work, produces blank screen. But if I scale to 0,75 with HiDPI it seems to work most of the times. Even if scaling to 1,0 it then comes out black. Seems that it needs some post-production in order to not be black... very strange. Hit me up if you need logs.
